dvc/output.py

```diff
@@ -523,14 +527,27 @@ def get_entry(self) -> "DataIndexEntry":
         except NoRemoteError:
             remote = None
 
-        return DataIndexEntry(
+        if self.files and not self.obj:
+            self.obj = self.get_obj()
+
+        entry = DataIndexEntry(
             meta=self.meta,
             obj=self.obj,
             hash_info=self.hash_info,
             odb=self.odb,
             cache=self.odb,
             remote=remote,
         )
+        if self.stage.is_import and not self.stage.is_repo_import:
+            dep = self.stage.deps[0]
+            entry.fs = dep.fs
+            entry.path = dep.fs_path
+            entry.meta = dep.meta
+            if not entry.obj:
+                if not dep.obj:
+                    dep.obj = dep.get_obj()
+                entry.obj = dep.obj
+        return entry
```

Comment:
still undecided if this function belongs in `output.get_entry` or if we need something like `repo.index.build()` that takes an output or stage and generates index entries